### PR TITLE
CRUD actions on quality profiles

### PR DESCRIPTION
**Pull request by jblievremont**
This PR will add several CRUD actions, starting with /delete

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java [hunk 1]
```diff
@@ -117,6 +117,7 @@ void delete(DbSession session, String key, boolean force) {
   }
 
   private void doDelete(DbSession session, QualityProfileDto profile) {
+    db.qualityProfileDao().deleteAllProjectProfileAssociation(profile.getKey(), session);
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java]: Just for my information, what has changed ?

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCopyAction.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.server.ws.WebService.NewAction;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.server.qualityprofile.QProfileCopier;
+import org.sonar.server.user.UserSession;
+
+public class QProfileCopyAction implements BaseQProfileWsAction {
+
+  private static final String PARAM_PROFILE_NAME = "toName";
+  private static final String PARAM_PROFILE_KEY = "fromKey";
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCopyAction.java]: What is the standard for parameter name ? It thought it was not camel case, but under case.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileParamUtils.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.WebService.NewAction;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.qualityprofile.db.QualityProfileDto;
+import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.server.qualityprofile.QProfileFactory;
+
+import static org.apache.commons.lang.StringUtils.isEmpty;
+
+public class QProfileParamUtils {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileParamUtils.java]: I think the name is not appropriate, as it's not related only to parameter.
- QProfileHelper ?
- QProfileUtils ?
I let you find a good name :)

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/QProfileCopyActionTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QProfileCopyActionTest {
+
+  private WsTester tester;
+
+  // TODO Replace with proper DbTester + EsTester medium test during removal of DaoV2
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/QProfileCopyActionTest.java]: I can't wait for this to happen !

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java]: Oh yes, now profile to projects associations are in a separate table.
So 2nd question, where are tests testing this ?

**Comment by julienlancelot** [comment]
Apply my feedback, and LGTM

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java (hunk 1 above)
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java]: https://github.com/SonarSource/sonarqube/pull/192/files#diff-bf9d12277b166585672f3288d726a80cR106

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCopyAction.java (hunk 2 above)
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCopyAction.java]: No, underscores are for actions, parameters use camel case.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileParamUtils.java (hunk 3 above)
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileParamUtils.java]: Well, it is about the declaration and handling of profile key/name related parameters. I'll try to find a more explicit name.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java]: Hum ok, so QProfileFactoryMediumTest will be removed ?

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCopyAction.java (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCopyAction.java]: ok

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java (hunk 1 above)
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java]: Yep, most probably tests will move during removal of DaoV2

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java]: ok

**Pull request closed without merging**